odoo/models.py

```diff
@@ -949,37 +949,61 @@ def load(self, fields, data):
         # list of (xid, vals, info) for records to be created in batch
         batch = []
         batch_xml_ids = set()
+        # models in which we may have created / modified data, therefore might
+        # require flushing in order to name_search: the root model and any
+        # o2m
+        creatable_models = {self._name}
+        for fs in fields:
+            if fs[0] in (None, 'id', '.id'):
+                continue
+            ffs = self._fields
+            if isinstance(ffs[fs[0]], odoo.fields.Many2one):
+                # this only applies for toplevel m2o (?) fields
+                if fs[0] in (self.env.context.get('name_create_enabled_fieds') or {}):
+                    creatable_models.add(ffs[fs[0]].comodel_name)
+            for f in fs:
+                if f in (None, 'id', '.id'):
+                    break
+
+                if isinstance(ffs[f], odoo.fields.One2many):
+                    creatable_models.add(ffs[f].comodel_name)
+                    ffs = self.env[ffs[f].comodel_name]._fields
```

Comment:
This is quite obscure.  At first sight I thought `ffs[f]` was wrong from the second element on.
I suggest to keep a variable with the *model* corresponding to `f`.
I suggest to use better variable names :wink: